dev-tools/mage/gotest.go

```diff
@@ -137,33 +138,43 @@ func DefaultTestBinaryArgs() TestBinaryArgs {
 // Use RACE_DETECTOR=true to enable the race detector.
 // Use MODULE=module to run only tests for `module`.
 func GoTestIntegrationForModule(ctx context.Context) error {
-	return RunIntegTest("goIntegTest", func() error {
-		module := EnvOr("MODULE", "")
-		if module != "" {
-			err := GoTest(ctx, GoTestIntegrationArgsForModule(module))
-			return errors.Wrapf(err, "integration tests failed for module %s", module)
+	module := EnvOr("MODULE", "")
+	modulesFileInfo, err := ioutil.ReadDir("./module")
+	if err != nil {
+		return err
+	}
+
+	foundModule := false
+	for _, fi := range modulesFileInfo {
+		if !fi.IsDir() {
+			continue
 		}
+		if module != "" && module != fi.Name() {
+			continue
+		}
+		foundModule = true
 
-		modulesFileInfo, err := ioutil.ReadDir("./module")
+		// Set MODULE because only want that modules tests to run inside the testing environment.
+		os.Setenv("MODULE", fi.Name())
```

Comment:
It's important to set MODULE here because at the top level we want to execute docker or kubernetes unique per module. If we didn't add this here then every module test would run inside of the testing environment which we do not want. The test runner uses the MODULE from the running environment to pass into the underlying test environment using `-e` in the docker case and `env: {}` in the kubernetes case.

Reply:
Yes I have updated it so its passed in instead of set in the current process.